File: Back/Back/Models/Produto.cs
namespace Back.Models;

public class Produto : Entity
{
    public string Codigo { get; private set; }
    public string Nome { get; private set; }
    public decimal PrecoUnitario { get; private set; }
    public string ImagemUrl { get; private set; }

    public Produto(string codigo, string nome, decimal precoUnitario, string imagemUrl)
    {
        Codigo = codigo;
        Nome = nome;
        PrecoUnitario = precoUnitario;
        ImagemUrl = imagemUrl;
    }
}

File: Front/src/models/produto.ts
export class Produto {
  id?: number;
  codigo?: string;
  nome?: string;
  precoUnitario?: number;
  imagemUrl?: string;

  constructor(id?: number, codigo?: string, nome?: string, precoUnitario?: number, imagemUrl?: string) {
    this.id = id;
    this.codigo = codigo;
    this.nome = nome;
    this.precoUnitario = precoUnitario;
    this.imagemUrl = imagemUrl;
  }
}
